docs/client/session.md

```diff
@@ -0,0 +1,25 @@
+# Semantic conventions for session
+This document defines semantic conventions to apply to client-side applications when tracking sessions.
+
+A session is collection of Logs, Events, and Spans associated with a specific device or user within a specific period of time. 
+A session is represented by a unique identifier that is attached to Logs, Events, and Spans as an attribute.
+
+The primary purpose of sessions is to provide insight into the series of user actions or events that lead a critical or 
+interesting occurrence in the application's use. Sessions also provide a means to quantify overall application usage.
+
+
+## Attributes
+
+This attribute can be applied to all signals generated by an agent. It may be omitted from metrics, due to their aggregated nature. 
+
+| Attribute   | Type    | Description | Examples                             | Requirement level | 
+|-------------|---------|-------------|--------------------------------------|-------------------|
+| session.id  |  String | A GUID      | 00112233-4455-6677-8899-aabbccddeeff | required          |
+
+## How a session operates
+After a period of timeout, the `session.id` MUST be refreshed.
+The timeout period MUST be restarted when any Logs, Events, or Spans are recorded.
+The session timeout period CAN be customized.
+Default session timeout SHOULD be 30 minutes. This should only be done when the agent is configured, and MUST NOT be updated in the middle of a session.
+
+Maximum session length SHOULD NOT exceed 4 hours.
```

Comment:
I think this is too prescriptive too. The console app or IoT use case will definitely blow this out of the water.

Reply:
I think it's necessary to have some kind of maximum session length, just due to hardware constraints of recalling session data from whatever is storing it. It doesn't necessarily need to be 4 hours, this is just a place holder.  Distilling the purpose of a session: providing context to the large pool of data by constraining that data to a relevant period of time for a single user... if IoT devices have unlimited sessions, it sort of defeats the purpose. Maybe IoT doesn't need a session: you can get the same unlimited collection of data by constraining you data view to the device the data is generated from (e.g.: `device.id`)

Reply:
Fair enough. Maybe  we can say there is a max length, but that is determined by the application/SDK depending on its specific use case?

Reply:
I think that's reasonable. Maybe: 
> Implementers SHOULD implement a maximum session length applicable to their use-case